code/__HELPERS/_string_lists.dm

```diff
@@ -1,27 +1,40 @@
 #define pick_list(FILE, KEY) (pick(strings(FILE, KEY)))
+#define pick_list_replacements(FILE, KEY) (strings_replacement(FILE, KEY))
 
 var/global/list/string_cache
+//This is a hack to track which file is being processed for string subkey
+//replacements, yes it's ugly, I don't know an easier way though
+var/global/string_filename_current_key

```

Comment:
I don't have control over how the string_subkey_lookup is called, it's done by the regex datum's Replace method.